File: Android-Core/03-Components/ListView/AdapterAndHolder/03baseadapter/src/main/java/top/iqqcode/a03baseadapter/ThirdActivity.java
package top.iqqcode.a03baseadapter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class ThirdActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_third);

        TextView mTextView = findViewById(R.id.third_tv_view);
        AppInfo itemInfo = this.getIntent().getParcelableExtra("itemInfo");
        mTextView.setText("APP: " + itemInfo.getAppName());
    }
}

File: Kotlin/a-Basic/ActivityDemo/app/src/main/java/top/iqqcode/activitydemo/ThirdActivity.kt
package top.iqqcode.activitydemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import top.iqqcode.activitydemo.databinding.ActivityThirdBinding

class ThirdActivity : AppCompatActivity() {

    private lateinit var binding: ActivityThirdBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityThirdBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val date = intent.getStringExtra("date")
        binding.textView.text = date

        val user = intent.getParcelableExtra<User>("user")
        binding.textViewContext.text = user?.username + " " + user?.password.toString()
    }
}